base/client.jl

```diff
@@ -272,6 +272,15 @@ function exec_options(opts)
         invokelatest(Main.Distributed.process_opts, opts)
     end
 
+    # Maybe redefine bounds checking if requested
+    if JLOptions().check_bounds != 0
+        if JLOptions().check_bounds == 1
+            Core.eval(Main, :(Core.should_check_bounds(boundscheck::Bool) = true))
+        else
+            Core.eval(Main, :(Core.should_check_bounds(boundscheck::Bool) = false))
```

Comment:
Instead of doing this we could have a lib like:

```julia
module BaseInbounds

using PrecompileTools

@recompile_invalidations begin
    Core.should_check_bounds(boundscheck::Bool) = true
end

end
```

and load that here?

That definition does take a while to compile. Loading it is `329 ms` and on-disk size is `111M`.

Right now this doesn't quite seem to work due to:

```
┌ BaseInbounds [e8e3d1da-c89d-4456-81ea-eb3240fbf100]
│  WARNING: Method definition should_check_bounds(Bool) in module Core at boot.jl:855 overwritten in module BaseInbounds at /home/vchuravy/src/julia/stdlib/BaseInbounds/src/BaseInbounds.jl:6.
│    ** incremental compilation may be fatally broken for this module **
└ 
```

```
julia> using BaseInbounds

julia> f(A) = A[100]
f (generic function with 1 method)

julia> @code_typed f([1])
CodeInfo(
1 ─ %1 = Base.arrayref(true, A, 100)::Int64
└──      return %1
) => Int64
```

Reply:
> Instead of doing this we could have a lib like:

Yes, I mentioned that in the original post, but it needs some special loading logic